src/Main.gd

```diff
@@ -150,32 +157,48 @@ func enable_endless_mode():
 # background along with the player to make the map seem infinite
 func sync_player_background_y():
 	var y_delta = player_node.position.y - player_init_position.y
-	background_node.rect_position.y = background_init_position.y + y_delta
 
+func show_pad_ui():
+	pack_download_page_instance.show()
+
+func hide_pad_ui():
+	pack_download_page_instance.hide()
+
+func instantiate_pad_ui():
+	pack_download_page_instance = pack_download_page_scene.instance()
+	var pad_manager = get_node("/root/PlayAssetPackManager")	
```

Comment:
When can this fail? What happens if this fails? (Never is an ok answer, but am just curious.)

Reply:
This line would succeed if:
- PlayAssetPackManager has been added to the [autoload list](https://docs.godotengine.org/en/stable/getting_started/step_by_step/singletons_autoload.html)
- This current scene is a part of the [SceneTree](https://docs.godotengine.org/en/stable/getting_started/step_by_step/scene_tree.html) (so there exist a path from current node to /root/PlayAssetPackManager) 
If this fails, ```get_node()``` would return ```null``` and we would encounter this error later on in code:
<img width="996" alt="Screen Shot 2020-08-11 at 9 53 18 AM" src="https://user-images.githubusercontent.com/14120761/89912785-921c1200-dbb8-11ea-8693-df7233241527.png">
However, since the two requirements are always true(Main scene is alway connected to root node), this line in our case will never fail.

Reply:
Thanks for explaining!